app/frontend/src/components/Inputs/Calendar.tsx

```diff
@@ -145,55 +141,40 @@ function CalendarRow(props: CalendarRowProps) {
   );
 }
 
-function CalendarCell({ selected = false, ...props }: CalendarCellProps) {
-  const [isSelected, setIsSelected] = useState(selected);
-
+function CalendarCell({cell, setToSelect, setIsMouseDown, isMouseDown, data, setData, toSelect}: CalendarCellProps) {
+  const [next, nextdata, handleSelect] = useDragToSelectUnselect({ ...cell }, data, toSelect);
+  
   useEffect(() => {
-    props.onChange(isSelected);
-  }, [isSelected]);
-
-  function handleClick() {
-    setIsSelected(!isSelected);
+    setData(nextdata)
+  }, [next])
+
+  function mouseDown(e: any) {
```

Comment:
I think we would want to keep it to work on click and on drag.

Ideally, it should work just like the [when2meet calendar](https://www.when2meet.com/?17722526-XW6jr).

Reply:
@Aveline-art I just fixed the bug. Can you confirm that you're also not able to reproduce it? I know it's a rare bug, but if it's not popping up after a few tries, especially after you leave the calendar component when the mouse is down and come back to the cells, then I think we should be fine. Thoughts?